apps/classroomio-com/src/lib/Navigation/Navigation.svelte

```diff
@@ -41,12 +48,21 @@
       subtitle: 'Seamlessly collaborate with your students'
     }
   ];
+
+  onMount(() => {
+    const storedActiveLink = localStorage.getItem('activeLink');
+    if (storedActiveLink) {
+      activeLink = storedActiveLink;
+    }
+  });
```

Comment:
@Chifez Local storage shouldn't be the source of truth for the active state, the URL should be the source of truth.

I noticed you use onClick and then you set a string of the state. The problem with this is the following:

1. You've introduced JS on:click to a link tag
2. When we load the page in `/blog`, the activeLink will not be properly set.


What you should actually do is just check if a string matches the current page url.

Example:

```

$: activeLink = $page.url.pathname;
$: isSuperpowersActive = superpowers.some((sp) => activeLink.includes(`#${sp.key}`));

<a
  href="/blog"
  class:active={activeLink.startsWith('/blog')}
>
</a>
<a
  href="/pricing"
  class:active={activeLink.startsWith('/pricing')}
>
</a>
```

Note: `activeLink` will always update when the URL changes and you don't need any js to keep track of that or even localstorage.